src/App.jsx

```diff
@@ -14,70 +16,75 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState(goods[8] || '');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const clearSelection = () => {
+    setSelectedGood('');
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? `${selectedGood} is selected`
+          : 'No goods selected'
+        }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => clearSelection()}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isSelected = good === selectedGood;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            return (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={cn(
+                  { 'has-background-success-light': isSelected },
+                )
+                }
+              >
+                <td>
+                  <button
+                    data-cy={isSelected
+                      ? 'RemoveButton'
+                      : 'AddButton'
+                    }
+                    type="button"
+                    className={`button ${isSelected
+                      ? ' is-info'
+                      : ''
+                    }`}
+                    onClick={() => (
+                      isSelected
+                        ? clearSelection()
+                        : setSelectedGood(good)
+                    )}
```

Comment:
Better to move all the logic out of the callback into a separate handler for better readability

Reply:
> Better to move all the logic out of the callback into a separate handler for better readability

Agree